File: ABC/abc024/abc024_b 1.py
#50点解法
max_t = 1100000
open = [False] * max_t

N, T = map(int, input().split())
for _ in range(N):
    A = int(input())
    for t in range(A, A+T):
        open[t] = True

print(sum(open[t] == True for t in range(max_t)))

File: ABC/abc024/abc024_b 1.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    // 50点解法
    int max_t = 1100000, N, T, A, ans = 0;
    vector<bool> open(max_t, false);

    cin >> N >> T;
    for (int i = 0; i < N; i++) {
        cin >> A;
        for (int t = A; t < A + T; t++) {
            open[t] = true;
        }
    }

    for (int t = 0; t < max_t; t++) {
        if (open[t] == true) ans++;
    }
    cout << ans << endl;
}